lib/src/screens/home/feedTab.dart

```diff
@@ -12,8 +24,140 @@ class FeedTab extends StatefulWidget {
+      child: Card(
+        color: MediaQuery.of(context).platformBrightness == Brightness.dark
+            ? Colors.grey[900]
+            : Colors.grey[300],
+        child: InkWell(
+          onTap: () => Navigator.of(context).push(
+            platformPageRoute(
+              context: context,
+              builder: (context) {
+                return _getScreenForFeedItem(item);
+              },
+            ),
+          ),
+          child: Container(
+            padding: EdgeInsets.all(5),
+            child: Column(
+              children: [
+                Row(
+                  children: [
+                    InkWell(
+                      onTap: () => Navigator.of(context).push(
+                        platformPageRoute(
+                          context: context,
+                          builder: (context) {
+                            return SectionView(
+                              sectionName: item.owner,
+                              sectionCode: item.itemCode,
+                            );
+                          },
+                        ),
+                      ),
+                      child: Text('${item.owner}'),
+                    ),
+                    Text(' posted ${item.name}'),
+                    Spacer(),
+                    Text('${postDateFormatter.format(item.postDate)}',
+                        style: TextStyle(color: Colors.grey[400])),
+                  ],
+                ),
+                SizedBox(height: 10),
+                Text(
+                  item.body,
+                  textAlign: TextAlign.center,
+                ),
+                SizedBox(height: 20),
+                Row(
+                  children: [
+                    if (item.type == FeedItemType.Homework)
+                      Row(
+                        children: [
+                          Icon(PlatformIcons(context).time),
+                          SizedBox(width: 5),
+                          Text('Appears in calendar',
+                              style: TextStyle(color: Colors.grey[450]))
+                        ],
+                      ),
+                    Spacer(),
+                    Button(
+                      onPressed: () => Navigator.of(context).push(
+                        platformPageRoute(
+                          context: context,
+                          builder: (context) {
+                            return _getScreenForFeedItem(item);
+                          },
+                        ),
+                      ),
+                      child: Row(
+                        children: [
+                          Icon(PlatformIcons(context).eyeSolid),
+                          SizedBox(width: 5),
+                          PlatformText('View'),
+                        ],
+                      ),
+                    ),
+                  ],
+                )
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
```

Comment:
This is a sign that some of this should ideally be refactored out into smaller, simpler views.